python/decorators.py

```diff
@@ -0,0 +1,59 @@
+import time
+
+def timer_decorator(func):
+    def wrapper(*args, **kwargs):
+        start_time = time.time()
+        result = func(*args, **kwargs)
+        end_time = time.time()
+        execution_time = end_time - start_time
+        print(f"Execution time: {execution_time} seconds")
+        return result
+    return wrapper
+
+
+def timer_decorator_v2(output: str = 'stdout'):
+    def decorator(func):
+        def wrapper(*args, **kwargs):
+            start_time = time.time()
+            result = func(*args, **kwargs)
+            end_time = time.time()
+            execution_time = end_time - start_time
+            if output == 'stdout':
+                print(f"Execution time: {execution_time} seconds")
+            elif output.endswith('.txt') or output.endswith('.log'):
+                with open(output, 'a') as file:
+                    file.write(f"Execution time: {execution_time} seconds\n")
+            return result
+        return wrapper
+    return decorator
+
+def cache_decorator(size: int = None):
+    cache = {}
+    def decorator(func):
+        def wrapper(*args, **kwargs):
+            key = args
```

Comment:
The current version doesn't take `kwargs` into account, which means:
```python
@cache_decorator(5)
def foo(x: float, y: float) -> float:
    return x + y

foo(1, y=1)  # not cached --> result: 2
foo(1, y=9)  # cached     --> result: 2
```

Note:
There's no reason for now to try to cache all possible parameter combinations correctly:
```
(1, 2) == (1, y=2) == (x=1, y=2)
```